storage/client/status.h

```diff
@@ -0,0 +1,88 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_STORAGE_CLIENT_STATUS_H_
+#define GOOGLE_CLOUD_CPP_STORAGE_CLIENT_STATUS_H_
+
+#include "storage/client/version.h"
+#include <iostream>
+
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+/**
+ * Wrap the status code and message returned by the underlying transport.
+ *
+ * This class is modeled after `grpc::Status`, it contains the status code and
+ * error message (if applicable) from a JSON request.
+ *
+ * TODO(#548) - define well-known error codes.
+ */
+class Status {
```

Comment:
Is this going to be a `storage`-specific `Status` class, or will this be used throughout this repo?

So many `Status` classes, so little time:

* https://github.com/google/protobuf/blob/master/src/google/protobuf/stubs/status.h
* https://github.com/google/google-api-cpp-client/blob/master/src/googleapis/client/util/status.h
* I'm sure there are more...

Reply:
> Is this going to be a storage-specific Status class, or will this be used throughout this repo?

For libraries that use gRPC, `grpc::Status` is a better choice.  For libraries that do not, this would be a better choice.  One alternative is to link against gRPC just for `grpc::Status`, don't like it, but maaaaaybe?

> So many `Status` classes, ...

sigh, yes.